titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/pod/DefaultPodAffinityFactory.java

```diff
@@ -247,11 +252,26 @@ private void processZoneConstraints() {
                 return;
             }
 
-            // If there is an EBS volume, it defaults to placement in the volume's zone
-            Optional<EbsVolume> optionalEbsVolume = EbsVolumeUtils.getEbsVolumeForTask(job, task);
-            if (optionalEbsVolume.isPresent()) {
-                addNodeAffinitySelectorConstraint(KubeConstants.NODE_LABEL_ZONE, optionalEbsVolume.get().getVolumeAvailabilityZone(), true);
-                return;
+            // If there is a Static IP allocation, get its zone
+            Optional<String> optionalIpAllocationZone = IpAddressAllocationUtils.getIpAllocationZoneForTask(job.getJobDescriptor(), task, titusRuntime.getCodeInvariants());
+
+            // If there is an EBS volume, get its zone
+            Optional<String> optionalEbsVolumeZone = EbsVolumeUtils.getEbsVolumeForTask(job, task)
+                    .map(EbsVolume::getVolumeAvailabilityZone);
+
+            if (optionalIpAllocationZone.isPresent() && optionalEbsVolumeZone.isPresent()) {
```

Comment:
We should prevent this case to happen in the job service. Are there some checks there already?

Reply:
Discussed offline and will add validation in a subsequent PR.